examples/gwlb/main.tf

```diff
@@ -0,0 +1,142 @@
+resource "azurerm_resource_group" "this" {
+  name     = "${var.name_prefix}${var.resource_group_name}"
+  location = var.location
+}
+
+module "vnet" {
+  for_each = var.vnets
+  source   = "../../modules/vnet"
+
+  virtual_network_name = "${var.name_prefix}${each.value.name}"
+  location             = var.location
+  resource_group_name  = azurerm_resource_group.this.name
+  address_space        = each.value.address_space
+
+  network_security_groups = each.value.network_security_groups
+  subnets                 = each.value.subnets
+
+  tags = var.tags
+}
+
+module "gwlb" {
+  source = "../../modules/gwlb"
+
+  name                = "${var.name_prefix}${var.gwlb.name}"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.this.name
+
+  backends = var.gwlb.backends
+
+  frontend_ip_config = {
```

Comment:
check also possibility to do outgoing traffic with GWLB:

* just sending outgoing traffic
* utilising the outgoing rules

Reply:
Added outbound traffic inspection using outbound rules.